src/main/drivers/vtx_common.c

```diff
@@ -163,4 +221,23 @@ bool vtxCommonGetDeviceCapability(vtxDeviceCapability_t *pDeviceCapability)
     memcpy(pDeviceCapability, &vtxDevice->capability, sizeof(vtxDeviceCapability_t));
     return true;
 }
+
+vtxSettingsConfig_t vtxCommonGetSettings(void)
+{
+    return (vtxSettingsConfig_t) {
+        .band = vtxSettingsConfig()->band,
+        .channel = vtxSettingsConfig()->channel,
+        .freq = vtxSettingsConfig()->freq,
+        .power = vtxSettingsConfig()->power,
+    };
+}
```

Comment:
This function returns a value on the stack and so will cause hard to track down bugs. Instead pass in a pointer:
```
void vtxCommonGetSettings(vtxSettingsConfig_t *vtxSettingsConfig)
{
    vtxSettingsConfig->band = vtxSettingsConfig()->band;
...etc
```

However I'm not sure why this function is required at all, since `vtxSettingsConfig()` could be used in its place.

Reply:
Agree with the pointer.  The use if the get/update was to facilitate simplified change capture (ie memcmp vs local declaration and check of each value individually) and to improve readability. I'll see if I can find a different way.